cdc/split.cc

```diff
@@ -633,21 +638,25 @@ static bool entries_match_row_state(const schema_ptr& base_schema, const cell_ma
         auto current_raw_map = cdef.type->deserialize(*it->second);
         map_type_impl::native_type current_values = value_cast<map_type_impl::native_type>(current_raw_map);
 
-        if (current_values.size() != e.cells.size()) {
+        if (current_values.size() != update.cells.size()) {
             return false;
         }
         
-        std::unordered_map<sstring_view, bytes> kvm;
+        std::unordered_map<sstring_view, bytes> current_values_map;
         for (const auto& entry : current_values) {
             const sstring& attr_name = value_cast<sstring>(entry.first);
             auto v = value_cast<bytes>(entry.second);
-            kvm[std::string_view(attr_name)] = v;
+            current_values_map[std::string_view(attr_name)] = v;
         }
 
-        for (auto& [key, value] : e.cells) {
+        for (auto& [key, value] : update.cells) {
             auto upd = value.value().linearize();
             auto k = to_string_view(key);
-            if (kvm[k] != upd) {
+            if (value.is_dead(now)) {
```

Comment:
I'm not sure `is_dead` should be used here. I think you should use `is_live()` instead (the 0 argument version). The `is_dead(gc_clock::time_point now)` overload should be used if you have some long-existing mutations which might expire due to their TTL. The preimage is fetched via our reading layer which, I believe, already filters out expired data. The rest of our CDC code that generates mutations does not care for it, so I don't think we should care about it here either.